### PR TITLE
CBAPI-4931: Mark CBCloudAPI.get_notifications() as deprecated

### PR DESCRIPTION
**Pull request by abowersox-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x] A self-review of the code has been done.

## What is the ticket or issue number?

https://jira.carbonblack.local/browse/CBAPI-4931

## Pull Request Description

Marked the `get_notifications()` and `notification_listener()` functions on `CBCloudAPI` as deprecated, and did a small "beauty pass" through the docstrings in that file.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Primarily doc changes, no substantive code changes. Unit tests still pass.

### CODE REVIEW

File: src/cbc_sdk/rest_api.py [hunk 1]
```diff
@@ -146,39 +146,52 @@ def audit_remediation_history(self, query=None):
...
         return res.get("notifications", [])
 
     def get_auditlogs(self):
-        """Retrieve queued audit logs from the Carbon Black Cloud Endpoint Standard server.
+        """
+        Retrieve queued audit logs from the Carbon Black Cloud Endpoint Standard server.
 
-        Notes:
+        Note:
             This can only be used with a 'API' key generated in the Carbon Black Cloud console.
```
Comment (avanbrunt-cb) [inline src/cbc_sdk/rest_api.py:200]: This is not true anymore as we support custom keys too now

**Review by avanbrunt-cb — APPROVED** [review]
One change otherwise I approve

File: src/cbc_sdk/rest_api.py (hunk 1 above)
Comment (kebringer-cb) [reply src/cbc_sdk/rest_api.py:200]: Could 
- the API type key be marked deprecated
- required permission is org.audits - READ

Comment (abowersox-cb) [reply src/cbc_sdk/rest_api.py]: Addressed this.

**Review by kebringer-cb — APPROVED** [review]
Thanks

**Review by emitreva-cb — APPROVED** [review]

**Pull request merged**